### ISSUE TITLE
make cross pollination work again

### ISSUE THREAD

**Issue by oliverchang** [google/oss-fuzz#1022]
Recent refactoring broke this.

**Comment by inferno-chromium** [comment]
See also https://github.com/google/oss-fuzz/issues/395, i think that can be tracked here as well.

**Comment by inferno-chromium** [comment]
This has now landed (@kcc as fyi), be ready for the roller coaster ride. Every day, we will be merging 3 random corpuses from current list of targets (~1500). Coverage related improvements and optimizations will come next as part of https://github.com/google/oss-fuzz/issues/1581

**Comment by alex** [comment]
@inferno-chromium Sorry, are you saying that you'll be taking 3 random corpuses from all current targets, and trying those on all other targets?

**Comment by inferno-chromium** [comment]
Sorry, let me rephrase, corpuses from 3 random fuzz targets (out of the list of 1500) and these corpuses will come from backup corpuses (90-days old).

**Comment by alex** [comment]
I still don't understand. Maybe some psuedo code :-)

```py
test_corpuses = [target.backup_corpus for target in random.sample(all_targets, 3)]
for target in all_targets:
    for test_corpus in test_corpuses:
        target.test_merge_corpus(test_corpus)
```

?

**Comment by inferno-chromium** [comment]
Correct!

and those test corpuses are not from current corpus, but instead corpus that was backup 90 days ago (to avoid catching recent regressions derived from that corpus, just too much of a caution atm).

**Comment by alex** [comment]
Thanks.

**Comment by inferno-chromium** [comment]
E.g. libFuzzer_ffmpeg_AV_CODEC_ID_WS_SND1_fuzzer  - MERGE-OUTER: 1 new files with 3 new features added
libFuzzer_graphicsmagick_coder_PNG_fuzzer - MERGE-OUTER: 2 new files with 2 new features added
etc
so seems to work, lets see if in next 2-3 day, we see a spike in coverage/bugs.

**Comment by alex** [comment]
Which corpus was it that was merged into those two targets?

**Comment by inferno-chromium** [comment]
Had to dig logs.

for graphicsmagick one, i see these two corpuses used
gs://imagemagick-backup.clusterfuzz-external.appspot.com/corpus/libFuzzer/imagemagick_encoder_ttf_fuzzer/imagemagick_encoder_ttf_fuzzer_minimized_corpus-2018-04-30.zip successfully unpacked into shared corpus. 
gs://graphicsmagick-backup.clusterfuzz-external.appspot.com/corpus/libFuzzer/graphicsmagick_coder_MONO_fuzzer/graphicsmagick_coder_MONO_fuzzer_minimized_corpus-2018-04-30.zip
